static/js/pages/SignUpPage/SignUpForm.jsx

```diff
@@ -37,67 +37,69 @@ class SignUpForm extends Component {
             },
             terms: {
                 value: '',
-                error: 'Required'
-            }
+                error: 'Required',
+            },
         };
 
         this.handleChange = this.handleChange.bind(this);
         this.handleSubmit = this.handleSubmit.bind(this);
         this.checkEmailExists = this.checkEmailExists.bind(this);
     }
 
+    setFieldValue(name, value) {
+        this.setState(Object.assign(this.state, {
```

Comment:
what are You doing? :)

https://facebook.github.io/react/docs/state-and-lifecycle.html#do-not-modify-state-directly